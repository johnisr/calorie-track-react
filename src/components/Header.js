import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const Header = ({ startLogout }) => (
  <header>
    <h1>Calorie-Track</h1>
    <NavLink to='/FoodDashboard' >Food dashboard</NavLink>
    <button onClick={startLogout}>Logout</button>
  </header>
);

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout()),
});

export default connect(undefined, mapDispatchToProps)(Header);