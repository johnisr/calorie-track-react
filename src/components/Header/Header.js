import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import moment from 'moment';
import { defaultLog, startAddLog } from '../../actions/logs';
import { addCurrentEditLog } from '../../actions/currentLog';
import { startLogout } from '../../actions/auth';

import './Header.css';


export class Header extends React.Component {
  handleEdit = () => {
    const today = moment().startOf('day').valueOf();
    const log = this.props.logs.filter(log => log.date === today)[0];
    if (!log) {
      const todayLog = {
        ...defaultLog,
        date: today,
      };
      this.props.startAddLog(todayLog);
      this.props.addCurrentEditLog(todayLog);
    } else {
      this.props.addCurrentEditLog(log);
    }
    this.props.history.push('/editlog');
    
  }
  render() {
    return (
      <header className="header">
        <h1 className="heading-primary header__title">Calorie-Track</h1>
        <NavLink className="btn" to='/Foods' >Foods</NavLink>
        <NavLink className="btn" to='/Logs' >Logs</NavLink>
        <NavLink className="btn" to='/info' >Info</NavLink>
        <button className="btn" onClick={this.handleEdit}>Today</button>
        <button className="btn flex-right-end" onClick={this.props.startLogout}>Logout</button>
      </header>
    );
  };
}

const mapStateToProps = (state) => ({
  logs: state.logs,
});

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout()),
  startAddLog: (log) => dispatch(startAddLog(log)),
  addCurrentEditLog: (log) => dispatch(addCurrentEditLog(log)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);