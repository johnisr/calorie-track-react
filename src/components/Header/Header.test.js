import React from 'react';
import { shallow } from 'enzyme';
import { Header } from './Header';

it('should render Header correctly', () => {
  const wrapper = shallow(<Header startLogout={() => {}}/>);
  expect(wrapper).toMatchSnapshot();
});

it('should call startLogout on button click', () => {
  const startLogout = jest.fn();
  const wrapper = shallow(<Header startLogout={startLogout}/>);
  wrapper.find('button').at(1).simulate('click');
  expect(startLogout).toHaveBeenCalled();
});