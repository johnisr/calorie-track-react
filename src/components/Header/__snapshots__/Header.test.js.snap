// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    startLogout={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="heading-primary header__title"
>
          Calorie-Track
</h1>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="btn"
          to="/Foods"
>
          Foods
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="btn"
          to="/Logs"
>
          Logs
</NavLink>,
        <button
          className="btn"
          onClick={[Function]}
>
          Today
</button>,
        <button
          className="btn flex-right-end"
          onClick={[Function]}
>
          Logout
</button>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Calorie-Track",
          "className": "heading-primary header__title",
        },
        "ref": null,
        "rendered": "Calorie-Track",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Foods",
          "className": "btn",
          "to": "/Foods",
        },
        "ref": null,
        "rendered": "Foods",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Logs",
          "className": "btn",
          "to": "/Logs",
        },
        "ref": null,
        "rendered": "Logs",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Today",
          "className": "btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Today",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Logout",
          "className": "btn flex-right-end",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Logout",
        "type": "button",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="heading-primary header__title"
>
            Calorie-Track
</h1>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="btn"
            to="/Foods"
>
            Foods
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="btn"
            to="/Logs"
>
            Logs
</NavLink>,
          <button
            className="btn"
            onClick={[Function]}
>
            Today
</button>,
          <button
            className="btn flex-right-end"
            onClick={[Function]}
>
            Logout
</button>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Calorie-Track",
            "className": "heading-primary header__title",
          },
          "ref": null,
          "rendered": "Calorie-Track",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Foods",
            "className": "btn",
            "to": "/Foods",
          },
          "ref": null,
          "rendered": "Foods",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Logs",
            "className": "btn",
            "to": "/Logs",
          },
          "ref": null,
          "rendered": "Logs",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Today",
            "className": "btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Today",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Logout",
            "className": "btn flex-right-end",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Logout",
          "type": "button",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
